protocols/v2/codec-sv2/examples/encrypted.rs

```diff
@@ -0,0 +1,167 @@
+        .get_payload_when_handshaking()
+        .try_into()
+        .expect("Handshake remote invlaid message");
+
+    let (second_message, receiver_state) = receiver_state
+        .step_1(first_message)
+        .expect("Responder failed second step of handshake");
+    let second_message: [u8; INITIATOR_EXPECTED_HANDSHAKE_MESSAGE_SIZE] = second_message
+        .get_payload_when_handshaking()
+        .try_into()
+        .expect("Handshake remote invlaid message");
+
+    let sender_state = sender_state
+        .step_2(second_message)
+        .expect("Initiator failed third step of handshake");
+
+    let mut sender_state = match sender_state {
+        State::Transport(c) => State::with_transport_mode(c),
+        _ => panic!("todo"),
+    };
+
+    let mut receiver_state = match receiver_state {
+        State::Transport(c) => State::with_transport_mode(c),
+        _ => panic!("todo"),
+    };
+
+    // Create a message
+    let nonce = 1337;
+    let msg = CustomMessage { nonce };
+    let msg_type = CUSTOM_MSG_TYPE;
+    // Unique identifier of the extension describing the protocol message, as defined by Sv2 Framing
+    let extension_type = 0;
+    // This message is intended for the receiver, so set to false
+    let channel_msg = false;
+
+    let frame = StandardEitherFrame::<CustomMessage>::Sv2(
+        Sv2Frame::from_message(msg, msg_type, extension_type, channel_msg)
+            .expect("Failed to create the frame"),
+    );
+
+    let mut encoder = NoiseEncoder::<CustomMessage>::new();
+    let encoded_frame = encoder
+        .encode(frame, &mut sender_state)
+        .expect("Failed to encode the frame");
+
+    // Send the encoded frame
+    stream_sender
+        .write_all(encoded_frame.as_slice())
+        .expect("Failed to send the encoded frame");
+
+    // Receiver side
+    let mut decoder = StandardNoiseDecoder::<CustomMessage>::new();
+    let decoder_buf = decoder.writable();
+    // Read the frame header into the decoder buffer
+    stream_receiver
+        .read_exact(decoder_buf)
+        .expect("Failed to read the encoded frame header");
+
+    // This returns a `MissingBytes` error because it only reads the header as there is no payload
+    // in memory yet. Therefore, we safely ignore the error as the important thing here is that we
+    // loaded the `decoder.missing_b` with the expected frame payload size
+    let _ = decoder.next_frame(&mut receiver_state);
+
+    // Now the decoder buffer has the expected size of the frame payload
+    let decoder_buf = decoder.writable();
+
+    // Read the payload into the decoder buffer
+    stream_receiver
+        .read_exact(decoder_buf)
+        .expect("Failed to read the encoded frame payload");
+
+    // INCORRECT PANIC HERE (DO NOT MERGE):
+    // Decode the frame
+    let _decoded_frame = decoder
+        .next_frame(&mut receiver_state)
+        .expect("Failed to decode the frame");
```

Comment:
This panics with:

```sh
thread 'main' panicked at v2/codec-sv2/examples/encrypted.rs:156:10:
Failed to decode the frame: MissingBytes(18)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Finished running. Exit status: 101]
```

@Shourya742, are you able to identify if the noise protocol has any errors that would cause this panic?

@fi3, are you able to identify any errors in the codec and framing related logic that would cause this panic?

Reply:
@fi3 you are correct. The examples have been updated to read from the TCP stream in a loop. This fixed the panic. 

However, when I run the examples, they work. But @plebhash has pulled down the code and ran them and they fail with [this stderr ](https://github.com/user-attachments/files/17297047/message.1.txt).

Are you able to run them successfully? Any idea has to why the errors are occurring?

Reply:
Nvm. I fixed with with appropriate usage of `#[cfg(feature = noise_sv2)]` decorators.